reminder-backend/infra/src/main/scala/me/kerfume/infra/impl/session/AuthenticationService.scala

```diff
@@ -0,0 +1,61 @@
+package me.kerfume.infra.impl.session
+
+import cats.effect.IO
+import me.kerfume.infra.impl.session.TwitterOAuthSessions.Session
+import me.kerfume.twitter.oauth.OAuthClient
+import cats.instances.option._
+import cats.syntax.traverse._
+import me.kerfume.twitter.oauth.OAuthClient.OAuthToken
+import sttp.model.Uri
+
+class AuthenticationService(
+    sessions: TwitterOAuthSessions,
+    twitterOAuthClient: OAuthClient
+) {
+  import AuthenticationService._
+
+  def twitterAuthenticationInit(
+      sessionKey: Option[String],
+      callBack: Uri
+  ): IO[TwitterSessionInitResult] = {
+    for {
+      sessionOpt <- sessionKey.traverse(sessions.getSession).map { _.flatten }
+      res <- sessionOpt match {
+        case Some(s) => IO { SessionExists(s) }
+        case None =>
+          for {
+            reqToken <- twitterOAuthClient.getRequestToken(callBack)
+            redirectUrl = twitterOAuthClient.redirectUrl(reqToken)
+            _ <- sessions.setPreSession(reqToken)
+          } yield {
+            NeedGetTwitterOAuthToken(redirectUrl, reqToken)
+          }
+      }
+    } yield res
+  }
+
+  def createSession(
+      tokenKey: String,
+      verifier: String
+  ): IO[Either[String, String]] = {
+    for {
+      token <- sessions.getPreSession(tokenKey)
+      res <- token match {
+        case Some(t) =>
+          for {
+            accessToken <- twitterOAuthClient.getAccessToken(t, verifier)
+            profile <- twitterOAuthClient.getProfile(accessToken)
+            key <- sessions.setSession(profile.id.toString, accessToken)
+          } yield Right(s"REMINDER_SESSION=$key")
```

Comment:
return only session Key this function.
attach cookie key by EndPoint or Controller.